### ISSUE TITLE
AWS ES domain advanced options not respected

### ISSUE THREAD

**Issue by chiefy** [hashicorp/terraform#4498]
`aws-es.tf`
```
resource "aws_elasticsearch_domain" "es" {
	depends_on = ["aws_iam_role.iam_for_lambda"]
	domain_name = "${var.es_domain}"
	advanced_options {
		"rest.action.multi.allow_explicit_index" = true
	}
...
```

After I apply my plan succesfully:
```bash
aws es describe-elasticsearch-domains --domain-names <my-domain-name>
{
    "DomainStatusList": [
        {
            "ElasticsearchClusterConfig": {
                "DedicatedMasterEnabled": false,
                "InstanceCount": 3,
                "ZoneAwarenessEnabled": false,
                "InstanceType": "t2.small.elasticsearch"
            },
            "Endpoint": "searchxxxx-xxxx.us-west-2.es.amazonaws.com",
            "Created": true,
            "Deleted": false,
            "DomainName": "odd-staging",
            "EBSOptions": {
                "VolumeSize": 10,
                "VolumeType": "gp2",
                "EBSEnabled": true
            },
            "SnapshotOptions": {
                "AutomatedSnapshotStartHour": 23
            },
            "DomainId": "xxx/xxxxx",
            "AccessPolicies": "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Effect\":\"Allow\",\"Principal\":{\"AWS\":[\"arn:aws:iam::xxx:root\",\"arn:aws:iam::xxx:user/xxx\"]},\"Action\":\"es:*\",\"Resource\":\"arn:aws:es:us-west-2:xxxx:domain/odd-catalog/*\"}]}",
            "Processing": false,
            "AdvancedOptions": {
                "rest.action.multi.allow_explicit_index": "false"
            },
            "ARN": "arn:aws:es:us-west-2:xxxx:domain/xxxx"
        }
    ]
} 
```

**Comment by radeksimko** [comment]
Closing as a duplicate of https://github.com/hashicorp/terraform/issues/3980
TL;DR Unfortunately this is a core bug related to dots in `TypeMap` keys.

Reopen if you think it's not a duplicate.

**Comment by chiefy** [comment]
Ah sorry @radeksimko I searched but obviously not good enough.
